x-pack/platform/plugins/shared/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -457,6 +460,8 @@ export function buildComponentTemplates(params: {
           : {}),
         dynamic_templates: mappingsDynamicTemplates.length ? mappingsDynamicTemplates : undefined,
         ...omit(indexTemplateMappings, 'properties', 'dynamic_templates', 'runtime'),
+        ...(isOtelInputType ? { dynamic: true } : {}),
+        ...(isOtelInputType && type === 'traces' ? { subobjects: undefined } : {}),
```

Comment:
We should override these values only if the package is not already configuring it. It is possible for packages to explicitly disable dynamic mappings, or use `dynamic: runtime`. These packages will need to handle possible conflicts with other settings such as `subobjects: false` by themselves.

See for example https://github.com/elastic/integrations/blob/60ac006d494fa97740c48c409a90b25dcac8527a/packages/kubernetes/data_stream/audit_logs/manifest.yml#L5-L7

Reply:
I pushed https://github.com/elastic/kibana/pull/235226/commits/0eb655e189c3de088490f0052ca6735d2ba82aee where I added a check to determine if the package has `runtime` fields or `dynamic: false` and in that case these overrides are not done.

Reply:
Take into account that it can be also `dynamic: runtime`, I think we shouldn't override it if it appear with any value.

Reply:
Fixed the condition: 
```
Object.keys(mappingsRuntimeFields).length > 0 || indexTemplateMappings?.dynamic !== undefined;
```